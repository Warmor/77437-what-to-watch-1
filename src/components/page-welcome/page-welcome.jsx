import React from 'react';
import PropTypes from "prop-types";
import {connect} from "react-redux";
import Footer from "~/components/footer/footer";
import Header from "~/components/header/header";
import MoviesList from "~/components/movies-list/movies-list";
import GenresList from "~/components/genres-list/genres-list";
import MovieCardInfo from "~/components/movie-card-info/movie-card-info";
import {getGenres} from "~/reducer/data/selectors";
import {getActiveFilms, getActiveGenre, getPromoFilm} from "~/reducer/catalog/selectors";
import {ActionCreator as ActionCatalog} from "~/reducer/catalog/catalog";

class PageWelcome extends React.PureComponent {

  render() {
    return <>
      <section className="movie-card">
        <div className="movie-card__bg">
          <img src={this.props.promoFilm && this.props.promoFilm.backgroundImage} alt={this.props.promoFilm && this.props.promoFilm.name}/>
        </div>
        <h1 className="visually-hidden">WTW</h1>
        <Header withAuth/>
        {(this.props.promoFilm && <MovieCardInfo film={this.props.promoFilm} withPoser/>)}
      </section>
      <div className="page-content">
        <section className="catalog">
          <GenresList
            genres={this.props.genres}
            activeGenre={this.props.activeGenre}
            onGenreLinkClick={this.props.setActiveGenre}/>
          <MoviesList films={this.props.films}/>
        </section>
        <Footer/>
      </div>
    </>;
  }
}

PageWelcome.propTypes = {
  films: PropTypes.arrayOf(PropTypes.object).isRequired,
  genres: PropTypes.array.isRequired,
  promoFilm: PropTypes.object,
  activeGenre: PropTypes.string.isRequired,
  setActiveGenre: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) => ({
  ...ownProps,
  films: getActiveFilms(state),
  activeGenre: getActiveGenre(state),
  genres: getGenres(state),
  promoFilm: getPromoFilm(state),
});

const mapDispatchToProps = (dispatch) => ({
  setActiveGenre: (genre) => dispatch(ActionCatalog.setActiveGenre(genre)),
});

export {PageWelcome};

export default connect(mapStateToProps, mapDispatchToProps)(PageWelcome);
