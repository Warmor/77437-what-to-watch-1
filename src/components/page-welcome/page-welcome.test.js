import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallowToJson} from 'enzyme-to-json';
Enzyme.configure({adapter: new Adapter()});

import {PageWelcome} from "~/components/page-welcome/page-welcome";

describe(`<PageWelcome/>`, () => {
  it(`PageWelcome renders correctly`, () => {
    const mockProps = {
      films: [{
        id: 1,
        title: `Fantastic Beasts`,
        genreId: 1,
        image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
        video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
      }],
      activeGenre: `Comedies`,
      genres: [`Comedies`],
      setActiveGenre: jest.fn(),
      loadFilmsAndGenre: jest.fn(),
    };
    const tree = shallow(<PageWelcome
      films={mockProps.films}
      genres={mockProps.genres}
      activeGenre={mockProps.activeGenre}
      setActiveGenre={mockProps.setActiveGenre}
      loadFilmsAndGenre={mockProps.loadFilmsAndGenre}/>);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
