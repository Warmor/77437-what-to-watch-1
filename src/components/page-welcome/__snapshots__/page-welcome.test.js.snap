// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageWelcome/> PageWelcome renders correctly 1`] = `
<Fragment>
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <ConnectFunction
      withAuth={true}
    />
  </section>
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <GenresList
        activeGenre="Comedies"
        genres={
          Array [
            "Comedies",
          ]
        }
        onGenreLinkClick={[MockFunction]}
      />
      <MoviesList
        films={
          Array [
            Object {
              "genreId": 1,
              "id": 1,
              "image": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
              "title": "Fantastic Beasts",
              "video": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
            },
          ]
        }
      />
    </section>
    <Footer />
  </div>
</Fragment>
`;
